src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,104 @@
-export const Pagination = () => {};
+import cn from 'classnames';
+
+type Props = {
+  items: string[],
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: React.Dispatch<React.SetStateAction<number>>,
+};
+
+export const Pagination = ({
+  items,
+  total,
+  perPage,
+  currentPage = 1,
+  onPageChange,
+}: Props) => {
+  const pages = [];
+
+  for (let i = 0; i < total; i += perPage) {
+    pages.push(items.slice(
+      i,
+      i + perPage < total ? i + perPage : total,
```

Comment:
looks really difficult too. I guess there should be a way to make it look simpler